### PR TITLE
salt and pepper noise class added

### PR DESCRIPTION
**Pull request by romberol**
#### Changes
Added Salt and Pepper noise class

Relates to #1977 

#### Type of change

- [x] 📚  Documentation Update
- [x] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/augmentation/_2d/intensity/salt_pepper_noise.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        self.flags = {"salt_prob": salt_prob, "pepper_prob": pepper_prob}
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        salt_prob = flags["salt_prob"]
+        pepper_prob = flags["pepper_prob"]
+        noisy_image = input.clone()
+        # Generate a random mask for salt pixels
+        salt_mask = torch.rand_like(input) < salt_prob
```
Comment (edgarriba) [inline kornia/augmentation/_2d/intensity/salt_pepper_noise.py:45]: i believe the sampling should happen in a separated function implementing a Generator: https://github.com/kornia/kornia/blob/master/kornia/augmentation/_2d/intensity/hue.py#L61

@shijianjian

File: kornia/augmentation/_2d/intensity/salt_pepper_noise.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        # Generate a random mask for pepper pixels
+        pepper_mask = torch.rand_like(input) < pepper_prob
+
+        # Set salt pixels to 1 (white)
+        noisy_image[salt_mask] = 1.0
```
Comment (edgarriba) [inline kornia/augmentation/_2d/intensity/salt_pepper_noise.py:51]: probably elementwise multiplication is faster than indexing

File: kornia/augmentation/_2d/intensity/salt_pepper_noise.py (hunk 1 above)
Comment (johnnv1) [reply kornia/augmentation/_2d/intensity/salt_pepper_noise.py:45]: +1

File: kornia/augmentation/_2d/intensity/salt_pepper_noise.py (hunk 2 above)
Comment (johnnv1) [reply kornia/augmentation/_2d/intensity/salt_pepper_noise.py:51]: example here https://github.com/kornia/kornia/pull/2609

**Review by johnnv1 — CHANGES_REQUESTED** [review]
also, need to add it into `docs/`

**Review by edgarriba — CHANGES_REQUESTED** [review]
this needs tests too

**Comment by edgarriba** [comment]
implemented in https://github.com/kornia/kornia/pull/2746

**Pull request closed without merging**